tests/phpunit/tests/test-upgrade.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+class Upgrade_Test extends PLL_UnitTestCase {
+	protected $options_backup;
+
+	public static function wpSetUpBeforeClass( WP_UnitTest_Factory $factory ) {
+		parent::wpSetUpBeforeClass( $factory );
+
+		self::create_language( 'en_US' );
+		self::create_language( 'fr_FR' );
+	}
+
+	/**
+	 * @ticket #1664 see {https://github.com/polylang/polylang-pro/issues/1664}.
+	 */
+	public function test_upgrade_to_3_4() {
+		wp_set_current_user( 1 );
+		update_user_meta( get_current_user_id(), 'pll_filter_content', 'en' );
+		$links_model = self::$model->get_links_model();
+		$admin = new PLL_Admin( $links_model );
+		$admin->init();
```

Comment:
To follow the same process as Polylang, this should be moved between `new PLL_upgrade()` and the `setup_theme` action. Aslo please create a new model.

Reply:
Missed that, done in 0374d8fe.